driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideEncryptionNotSpawnMongocryptdTest.java

```diff
@@ -0,0 +1,244 @@
+                .extraOptions(commonExtraAutoEncryptionOptions(mongocryptdSocketAddress))
+                .bypassQueryAnalysis(true)
+        ).insertOne(Document.parse("{unencrypted: 'test'}"));
+        assertMongocryptdNotSpawned();
+    }
+
+    private MongoCollection<Document> setUpCollection(
+            final BiConsumer<InetSocketAddress, AutoEncryptionSettings.Builder> autoEncryptionSettingsBuilderMutator) {
+        setUpKeyVaultNamespace();
+        InetSocketAddress localMongocryptdSocketAddress = new InetSocketAddress(
+                InetAddress.getLoopbackAddress(), findAvailableMongocryptdLoopbackPort());
+        AutoEncryptionSettings.Builder autoEncryptionSettingsBuilder = AutoEncryptionSettings.builder();
+        autoEncryptionSettingsBuilderMutator.accept(localMongocryptdSocketAddress, autoEncryptionSettingsBuilder);
+        mongocryptdSocketAddress = localMongocryptdSocketAddress;
+        client = createMongoClient(MongoClientSettings.builder(getMongoClientSettings())
+                .autoEncryptionSettings(autoEncryptionSettingsBuilder
+                        .kmsProviders(singletonMap(LOCAL_KMS_PROVIDER_ID, singletonMap("key", localKmsProviderKey())))
+                        .keyVaultNamespace(KEY_VAULT_NAMESPACE.getFullName())
+                        .schemaMap(singletonMap(NAMESPACE.getFullName(), EXTERNAL_SCHEMA))
+                        .build())
+                .build());
+        MongoDatabase db = client.getDatabase(NAMESPACE.getDatabaseName());
+        db.drop();
+        return db.getCollection(NAMESPACE.getCollectionName());
+    }
+
+    private void setUpKeyVaultNamespace() {
+        try (MongoClient client = createMongoClient(MongoClientSettings.builder().build())) {
+            MongoDatabase db = client.getDatabase(KEY_VAULT_NAMESPACE.getDatabaseName());
+            db.drop();
+            db.getCollection(KEY_VAULT_NAMESPACE.getCollectionName(), BsonDocument.class)
+                    .withWriteConcern(WriteConcern.MAJORITY)
+                    .insertOne(EXTERNAL_KEY);
+        }
+    }
+
+    private static Map<String, Object> commonExtraAutoEncryptionOptions(final InetSocketAddress mongocryptdSocketAddress) {
+        return singletonMap("mongocryptdSpawnArgs", asList(
+                // We pick a random available `mongocryptd` port and also include it in the PID file path
+                // to reduce the chances of different test runs interfering with each other. The interference
+                // may come from the fact that once spawned, `mongocryptd` stays up and running for some time,
+                // which may cause failures in other runs if they use the same `mongocryptd` port / PID file.
+                format("--pidfilepath=bypass-spawning-mongocryptd-%d.pid", mongocryptdSocketAddress.getPort()),
+                format("--port=%d", mongocryptdSocketAddress.getPort())));
```

Comment:
The comment in the code explains why this is done the way it is, even though the prose tests do not require anything like that.